File: src/test/java/examples/java/receiver/firstequalssecond/Incorrect0.java
package examples.java.receiver.firstequalssecond;

public class Incorrect0 extends SimpleArrayList {
  public boolean firstEqualsSecond() {
    if (values.length <= 1) {
      throw new IllegalArgumentException();
    }
    return values[0].equals(values[0]);
  }
}

File: src/test/java/examples/java/receiver/equalwithkotlin/Incorrect0.kt
package examples.java.receiver.equalwithkotlin

@Suppress("IntroduceWhenSubject", "EqualsOrHashCode", "EqualsWithHashCodeExist")
class Incorrect0(private val value: Int) {
    override fun equals(other: Any?) = when {
        other!!.javaClass == javaClass -> {
            other as Incorrect0
            other.value == value
        }
        else -> false
    }
}